### ISSUE TITLE
Fix "save session" widget and restore to app

### ISSUE THREAD

**Issue by torrin47** [Innovate-Inc/EnviroAtlas#467]
Core requirements:

- [ ] Layers added to the LayerList widget are saved and restored to that widget
- [ ] Custom symbology set for any layer is saved and restored
- [ ] Selected Community is saved and restored
- [ ] Map Extent is saved and restored
- [ ] Local data added via the "Add Data" widget is saved and restored

Anything else? Maybe some future widgets?

**Comment by Baohong** [comment]
Above are the issues to be addressed. I also found two more issues when I tested from my local:
1. After three or four sessions are saved, no new session could be saved. In the log file, there is message: "NS_ERROR_DOM_QUOTA_REACHED: Persistent storage maximum size reached". We need to explore this problem.
2. One other issue is the graphics saved with session. For example, in Elevation Profile tool, graphics of path could be saved with a session name and the sessions could be downloaded as a file. But when the downloaded file is loaded, profile result could not be recovered even though the graphics of path could be recovered. To recover the profile result would be very complicated. We need to address the graphics issue also.
I will update if more issues are found.
